rust/otap-dataflow/crates/engine/src/error.rs

```diff
@@ -38,9 +34,49 @@ pub enum Error<T> {
         node: NodeId,
 
         /// The error that occurred.
-        error: SendError<NodeControlMsg>,
+        error: SendError<T>,
     },
 
+    /// A type-less error in ErrorT<T> context.
+    #[error("{0}")]
+    Error(Error),
+}
+
+impl<T: Sized> From<ErrorT<T>> for Error {
+    /// This drops the SendError<T> field yielding an untyped error.
+    fn from(value: ErrorT<T>) -> Self {
+        match value {
+            ErrorT::ChannelSendError(_) => Error::ChannelSendError,
+            ErrorT::PipelineControlMsgError(_) => Error::PipelineControlMsgError,
+            ErrorT::NodeControlMsgSendError { node, .. } => Error::NodeControlMsgSendError(node),
+            ErrorT::Error(e) => e,
+        }
+    }
+}
+
+/// All errors that can occur in the pipeline engine infrastructure.
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    /// A wrapper for the config errors.
+    #[error("A config error occurred: {0}")]
+    ConfigError(#[from] Box<otap_df_config::error::Error>),
+
+    /// A wrapper for the channel errors.
+    #[error("A channel error occurred: {0}")]
+    ChannelRecvError(#[from] otap_df_channel::error::RecvError),
+
+    /// A wrapper for the channel errors.
+    #[error("A pipeline data channel error occurred")]
+    ChannelSendError,
```

Comment:
Erasing the `T` at this level is correct, but I think we should add a `reason` or `message` field of type string in which we would place the `to_string` of the initial error.

Reply:
Done.